File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/request/OrderRequest.java
package br.com.edu.order.infra.rest.v1.request;

import br.com.edu.order.domain.Item;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class OrderRequest {

    @JsonProperty("created_by")
    private String createdBy;

    private String origin;

    private List<ItemRequest> items;

    public OrderRequest() {
        items = new ArrayList<>();
    }

    public List<ItemRequest> getItems() {
        if (items == null) {
            items = new ArrayList<>();
        }
        return items;
    }

    public List<Item> toModelItems() {
        return getItems().stream().map(ItemRequest::toModelItem).toList();
    }
}

File: order.service/order-service-app/src/main/kotlin/br/com/edu/order/infra/rest/v1/request/OrderRequest.kt
package br.com.edu.order.infra.rest.v1.request

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import br.com.edu.order.domain.Item

@JsonIgnoreProperties(ignoreUnknown = true)
data class OrderRequest(

    @JsonProperty("created_by")
    val createdBy: String? = null,

    @JsonProperty("origin")
    val origin: String? = "",

    @JsonProperty("items")
    val items: List<ItemRequest>? = listOf()
) {

    fun toModelItems(): MutableList<Item> = items?.map { obj: ItemRequest -> obj.toModelItem() }?.toMutableList() ?: mutableListOf()
}
